checks/webhook.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	sce "github.com/ossf/scorecard/v4/errors"
+)
+
+const (
+	// CheckWebHooks is the registered name for WebHooks.
+	CheckWebHooks = "Webhooks"
+)
+
+//nolint:gochecknoinits
+func init() {
+	if err := registerCheck(CheckWebHooks, WebHooks, nil); err != nil {
```

Comment:
Could you flag-guard the `registerCheck` call using `SCORECARD_V6` env variable? We want to make sure we test it before releasing.

Reply:
should we do this in this flag or in https://github.com/ossf/scorecard/blob/504f13441636dcc6273d45d9dc1e7ea1ef080ae2/cmd/root.go#L317 (since we already have the function https://github.com/ossf/scorecard/blob/504f13441636dcc6273d45d9dc1e7ea1ef080ae2/cmd/root.go#L261)?

Reply:
did in the root.go, but can change if that is better outside

Reply:
is this good?
when is not set 
```

Reply:
It's probably best to flag guard this in the `webhook.go` file since without it the Scorecard cron job (in `cron` package) will start using this. You could separately disable it in `cron` too, but that's just more work for you and might be simpler if we flag guard it here.

Reply:
re: when is not set -> I think that's fine. It's unsupported until we remove the env variable guard, so it's fine to get the panic.

Reply:
i will update, thanks so much for the reviews and feeback

Reply:
I've implemented this @laurentsimon, let me know your thoughts

```console
$ ./scorecard --repo github.com/cpanato/testing-ci-providers --checks Webhooks
Starting [Webhooks]
|--------------------------------------------------------------------------|------------------|
|                              GITHUB WEBHOOK                              | USES AUTH SECRET |
|--------------------------------------------------------------------------|------------------|
| https://github.com/cpanato/testing-ci-providers/settings/hooks/289347313 | true             |
| https://github.com/cpanato/testing-ci-providers/settings/hooks/345108191 | false            |
|--------------------------------------------------------------------------|------------------|

Finished [Webhooks]

RESULTS
-------
Aggregate score: 5.0 / 10

Check scores:
|--------|----------|--------------------------------|---------------------------------------------------------------------------------------------------------|
| SCORE  |   NAME   |             REASON             |                                        DOCUMENTATION/REMEDIATION                                        |
|--------|----------|--------------------------------|---------------------------------------------------------------------------------------------------------|
| 5 / 10 | Webhooks | webhooks with no secrets       | https://github.com/ossf/scorecard/blob/477737b08abd2c33ca3ff76f50d7db84641db2dd/docs/checks.md#webhooks |
|        |          | configured detected -- score   |                                                                                                         |
|        |          | normalized to 5                |                                                                                                         |
|--------|----------|--------------------------------|---------------------------------------------------------------------------------------------------------|
```

Reply:
I don't see the flag-guard anymore, maybe this got removed due to recent refactoring? We need something like:

```
// This is psuedo-code
if env("SCORECARD_V6") == 1 {
  registerCheck(CheckWebhook);
}
```

Reply:
@azeemshaikh38 i did here https://github.com/ossf/scorecard/pull/1675/files#diff-a52e58522f73e84652650c20022c6e11f2935c7add0723d8d7322df9bb21e59cR42

I tried on the registercheck but that failed to compile, because we complain about 

that is why I move to another point. but if that is not correct let me know

```
panic: check present in checks.yaml is not part of `checks.AllChecks`: Webhooks

goroutine 1 [running]:
main.main()
        /Users/cpanato/code/src/github.com/ossf/scorecard/docs/checks/internal/validate/main.go:61 +0x268
exit status 2
```

Reply:
Makes sense, thanks!